try to fix pyjwt issue

cqc-melf opened the pull request.

cqc-melf commented:
try to fix https://github.com/CQCL/pytket/issues/153

cqc-melf commented:
Works locally, but I am not happy with this, would call this a temporary fix.

cqc-alec reviewed (changes requested):
I agree this is only a temporary fix. One problem is that `requirements.txt` is also used in the `build-manual` script to insert the information into the documentation; removing `pytket-pyquil` from it means it won't appear in the list of requirements in the docs.

I think we should keep `requirements.txt` as it was (apart from bumping the pytket version), but install the packages explicitly in the `build-manual` script, instead of the existing seddery. Unfortunately this means having to update them in both places, so there should be a comment in both places to say this.

cqc-melf commented:
I have added the comments and moved the installation, I have updated the pytket and pytket-quantinuum version, too. I have run all the checks locally with the new changes for the notebooks and the manual.

cqc-melf commented:
Even with this changes is still only a temporary fix in my opinion.

cqc-melf commented:
(do we have a task for the not running checks documented somewhere?)

cqc-alec commented on requirements.txt:
```diff
@@ -1,8 +1,12 @@
-    pytket == 1.4.1
+[comment]: # if version are updated, update the version in the build-manual file, too
```
Hmm. Both `check-examples.yml` and `check-manual.yml` do `pip install -r requirements.txt` so this won't work. They will both have to be changed. Perhaps we need an `install-requirements` script?

cqc-melf replied:
I have just seen that, too. I have just added the same installation process in the same way, having a script might be better... Will do that now.

cqc-alec commented on .github/workflows/check-examples.yml:
```diff
@@ -40,9 +40,13 @@ jobs:
         python-version: 3.8
     - name: install python requirements for notebooks
       run: |
+        # if version are updated, update the version in the build-manual file
+        # and the requirements.txt, too.
+        # this duplication is currently necessary because of an issue with the pyjwt
+        # version required from qcs-api-client < 0.21.0
```
This comment doesn't need to go here (since this file doesn't need updating).

cqc-melf replied:
Oh yes, forgot to update. Changed now.

cqc-alec commented on install-requirements.sh:
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Better to call this file just `install-requirements` (no suffix), since it is executable with a shebang.
https://google.github.io/styleguide/shellguide.html#file-extensions

cqc-melf replied:
done

cqc-alec commented on requirements.txt:
```diff
@@ -1,8 +1,13 @@
-    pytket == 1.4.1
+[comment]: # if version are updated, update the version in the build-manual file
+[comment]: # and the check-examples.yml, too.
```
Don't need to update `check-examples.yml`.

cqc-melf replied:
yes, thank you. Done

cqc-alec commented on requirements.txt:
```diff
@@ -1,8 +1,13 @@
+[comment]: # if version are updated, update the version in the build-manual file
+[comment]: # and the check-examples.yml, too.
+[comment]: # this duplication is currently necessary because of an issue with the pyjwt
+[comment]: # version required from qcs-api-client < 0.21.0
+
+    pytket == 1.4.3
     pytket-cirq == 0.25.0
-    pytket-projectq == 0.23.0
     pytket-pyquil == 0.25.0
+    pytket-projectq == 0.23.0
```
Is this change of order necessary? (They were alphabetical.)

cqc-melf replied:
No it is not, have ordered them again alphabetical

cqc-melf commented:
Thank you for all the comments, have updated hopefully everything!
